docs/common/headingManager.test.ts

```diff
@@ -16,19 +17,34 @@ describe('HeadingManager tests', () => {
 
   test('_findMetaForTitle not returning same title twice', () => {
     const TITLE = 'Some Title';
-    const meta = { headings: [{ title: TITLE, processed: true }] };
-    const headingManager = new HeadingManager(SluggerStub, meta);
+    const meta = { headings: [{ title: TITLE, _processed: true }] };
+    const headingManager = new HeadingManager(
+      SluggerStub,
+      (meta as unknown) as PageMetadata
+    ) as any;
 
-    const result = headingManager._findMetaForTitle(TITLE);
+    const result = headingManager.findMetaForTitle(TITLE);
     expect(result).toBeUndefined();
   });
+
+  test('_findMetaForTitle marks meta as processed', () => {
+    const TITLE = 'Some Title';
+    const meta = { headings: [{ title: TITLE }] };
+    const headingManager = new HeadingManager(
+      SluggerStub,
+      (meta as unknown) as PageMetadata
+    ) as any;
+
+    const result = headingManager.findMetaForTitle(TITLE);
+    expect(result._processed).toBeTruthy();
+  });
 });
 
 describe('HeadingManager.addHeading()', () => {
   const META_TITLE = 'Meta heading 1';
   const META_LEVEL = 3;
   const meta = { maxHeadingDepth: 3, headings: [{ title: META_TITLE, level: META_LEVEL }] };
-  const headingManager = new HeadingManager(SluggerStub, meta);
+  const headingManager = new HeadingManager(SluggerStub, meta as PageMetadata) as any;
```

Comment:
Do we need the last `as any`? It seems like counter intuitive for me, we lose the heading manager typing right?

Reply:
I don't like it, but I needed this to test private properties.